src/checkers/inference/model/serialization/ToStringSerializer.java

```diff
@@ -24,49 +28,65 @@
 import checkers.inference.model.VariableSlot;
 
 /**
- * This Serializer is meant only to convert constraints and variables to
- * human readable strings.  It is used currently by the DebugSolver.
+ * This Serializer is meant only to convert constraints and variables to human readable strings. It
+ * is used currently by the DebugSolver.
  */
 public class ToStringSerializer implements Serializer<String, String> {
     private final boolean showAstPaths;
     private boolean showVerboseVars;
 
-    private int indent = 0;
-    public static final String INDENT_STRING = "    ";
+    private int indentationLevel = 0;
+
+    // 2 spaces per indentation
+    public static final String INDENT = "  ";
+
+    // stores N concatenations of INDENT at index N, where index 0 stores empty string
+    // the index is the indentation level
+    protected final List<String> indentStrings = new ArrayList<>();
+
+    // used to format constant slots
+    protected final AnnotationFormatter formatter;
 
     public ToStringSerializer(boolean showAstPaths) {
         this.showAstPaths = showAstPaths;
         this.showVerboseVars = true;
+        formatter = InferenceMain.getInstance().getRealTypeFactory().getAnnotationFormatter();
+
+        // set first value to ""
+        indentStrings.add("");
     }
 
-    public void setIndent(int indent) {
-        this.indent = indent;
+    public void setIndentationLevel(int indentationLevel) {
+        // Ensure the indentation level is always >= 0
+        this.indentationLevel = indentationLevel > 0 ? indentationLevel : 0;
     }
 
-    public int getIndent() {
-        return indent;
+    private int getIndentationLevel() {
```

Comment:
If this method is private, it's only get called in current class. Then why not directly reference `indentationLevel` field?